### ISSUE TITLE
change the InterMineObject.id field from an Integer to a Long 

### ISSUE THREAD

**Issue by julie-sullivan** [intermine/intermine#533]

**Comment by julie-sullivan** [comment]
@justinccdev @JoeCarlson Richard looked at this previously and was not able to make this change.

**Comment by justinccdev** [comment]
Any thoughts on the alternative solution of setting the starting object id to be -2^31?  A quick Google doesn't show any obvious problem in principle, though some posts about data adapters relying on negative IDs for temporary purposes (e.g. http://stackoverflow.com/questions/2473215/negative-primary-keys).  Hopefully not the case with us.

But it sounds like something went wrong recently with negative IDs, as @JoeCarlson says on the mailing list [2].  Joe, perhaps you could quickly post what went wrong in the webapp?  It might possibly show if there's an easy fix, though I'm probably being optimistic.

Of course, this is kicking the can down the road but perhaps a better short-term solution than the significant engineering and test work of expanding to bigint.

[1] http://stackoverflow.com/questions/2473215/negative-primary-keys
[2] http://gmod.827538.n3.nabble.com/64-bit-object-id-s-tt4052116.html

**Comment by julie-sullivan** [comment]
@justinccdev Yes it did used to work with negative numbers. @kkarra just had a problem with this and the error message listed in #1114 is the HTML header JSP which sets the title of in the browser. 

```
13/08/15,10:54:02 WARN  org.intermine.web.struts.HtmlHeadController     - object ID not a number! -1122965846
```
That would be an easy fix but probably not @JoeCarlson's problem.

**Comment by JoeCarlson** [comment]
One thing I saw was using isNumeric to check for numerical object ids. (I'm not at my desk now and cannot recall the exact source file name .... HtmlHeaderSomething.java .... Or something)

This is false for anything other than a string of digits. Negative numbers return false. It looks like This bit of code was put in a year ago. 

I tried modifying this bit of code but the negative ids still did not work. There is another problem somewhere. 

After seeing this, I removed the objects with negative ids and was able to reload them with positive ids. All is good now, but the longer term problem remains. 

Joe

**Comment by julie-sullivan** [comment]
This is the other bit that should change too:

https://github.com/intermine/intermine/blob/dev/intermine/web/main/src/org/intermine/web/struts/ReportController.java#L248-L250

**Comment by justinccdev** [comment]
@JoeCarlson modifying the code led to another explicit error or something like a blank webpage?

**Comment by JoeCarlson** [comment]
and the case I had seen before was

https://github.com/intermine/intermine/blob/dev/intermine/web/main/src/org/intermine/web/struts/HtmlHeadController.java#L111-L116 <https://github.com/intermine/intermine/blob/dev/intermine/web/main/src/org/intermine/web/struts/HtmlHeadController.java#L111-L116>

**Comment by JoeCarlson** [comment]
I tried modifying HtmlHeadController. This eliminated the ‘object ID is not a number!’ message from the log file but I still got the ‘Object not found’ web page

like:

http://www.flymine.org/flymine/report.do?id=-1058901

Joe
